caches/cache-redis/src/main/java/org/eclipse/hono/deviceconnection/redis/RedisCache.java

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.deviceconnection.redis;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.hono.cache.Cache;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonObject;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+
+/**
+ * TODO.
+ * @param <K> TODO
+ * @param <V> TODO
+ */
+public class RedisCache<K, V> implements Cache<K, V> {
```

Comment:
Please ignore this implementation for now. It was only made as a proof-of-concept.

Future plan is to properly implement this using the infinispan implementation as a guide and to investigate using the [vertx-redis](https://vertx.io/docs/vertx-redis-client/java/) client.